src/main/java/com/denizenscript/depenizen/bukkit/objects/breweryx/BRecipeTag.java

```diff
@@ -0,0 +1,349 @@
+package com.denizenscript.depenizen.bukkit.objects.breweryx;
+
+import com.denizenscript.denizen.objects.ItemTag;
+import com.denizenscript.denizencore.objects.Fetchable;
+import com.denizenscript.denizencore.objects.ObjectTag;
+import com.denizenscript.denizencore.objects.core.ColorTag;
+import com.denizenscript.denizencore.objects.core.ElementTag;
+import com.denizenscript.denizencore.objects.core.ListTag;
+import com.denizenscript.denizencore.tags.Attribute;
+import com.denizenscript.denizencore.tags.TagContext;
+import com.dre.brewery.recipe.BEffect;
+import com.dre.brewery.recipe.BRecipe;
+import com.dre.brewery.recipe.RecipeItem;
+import com.dre.brewery.utility.Tuple;
+import org.bukkit.Color;
+import org.bukkit.Material;
+
+import java.util.Optional;
+
+public class BRecipeTag implements ObjectTag {
+
+    // <--[ObjectType]
+    // @name BRecipeTag
+    // @prefix brecipe
+    // @base ElementTag
+    // @format
+    // The identity format for brewery is <recipe_name>
+    // For example, 'brecipe@my_recipe'.
+    //
+    // @plugin Depenizen, BreweryX
+    // @description
+    // A BRecipeTag represents a Brewery recipe.
+    //
+    // -->
+
+
+    @Fetchable("brecipe")
+    public static BRecipeTag valueOf(String string, TagContext context) {
+        if (string.startsWith("brecipe@")) {
+            string = string.substring("brecipe@".length());
+        }
+
+        BRecipe recipe = BRecipe.get(string);
+        if (recipe == null) {
+            return null;
+        }
+        return new BRecipeTag(recipe);
+    }
+
+    public static boolean matches(String arg) {
+        arg = arg.replace("brecipe@", "");
+        return BRecipe.get(arg) != null;
+    }
+
+    BRecipe bRecipe;
+
+    public BRecipeTag(BRecipe bRecipe) {
+        this.bRecipe = bRecipe;
+    }
+
+    String prefix = "brecipe";
+
+    @Override
+    public String getPrefix() {
+        return prefix;
+    }
+
+    @Override
+    public boolean isUnique() {
+        return true;
+    }
+
+    @Override
+    public String identify() {
+        return "brecipe@" + bRecipe.getRecipeName();
+    }
+
+    @Override
+    public String identifySimple() {
+        return identify();
+    }
+
+    @Override
+    public ObjectTag setPrefix(String aString) {
+        prefix = aString;
+        return this;
+    }
+
+
+
+    @Override
+    public ObjectTag getObjectAttribute(Attribute attribute) {
+        if (attribute == null) {
+            return null;
+        }
+
+        // <--[tag]
+        // @attribute <BRecipeTag.id>
+        // @returns ElementTag
+        // @plugin Depenizen, BreweryX
+        // @description
+        // Returns the ID of the recipe as specified in the config.
+        // -->
+        if (attribute.startsWith("id")) {
```

Comment:
This is the legacy tag system, again look at modern code or https://github.com/DenizenScript/Denizen/issues/2355 or https://guide.denizenscript.com/guides/contributing/property-dev.html